src/dotnet/HQ.Server/Services/QuoteServiceV1.cs

```diff
@@ -17,6 +18,65 @@ public QuoteServiceV1(HQDbContext context)
             this._context = context;
         }
 
+        public async Task<Result<UpsertQuotestV1.Response>> UpsertQuoteV1(UpsertQuotestV1.Request request, CancellationToken ct = default)
+        {
+            using (var transaction = await _context.Database.BeginTransactionAsync(System.Data.IsolationLevel.Serializable, ct))
+            {
+                try
+                {
+                    var validationResult = Result.Merge(
+                        Result.FailIf(string.IsNullOrEmpty(request.Name), "Name is required.")
+                    );
+
+                    if (validationResult.IsFailed)
+                    {
+                        return validationResult;
+                    }
+
+                    var quote = await _context.Quotes.FindAsync(request.Id);
+                    if (quote == null)
+                    {
+                        quote = new Quote();
+                        _context.Quotes.Add(quote);
+                    }
+
+                    quote.ClientId = request.ClientId;
+                    quote.Name = request.Name;
+                    quote.Value = request.Value;
+                    quote.Date = request.Date;
+                    quote.Status = request.Status;
+
+
+                    var latestChargeCode = _context.Quotes.Max((q) => q.ChargeCode.Code);
+                    var latestQuoteNumber = int.Parse(latestChargeCode.Substring(1));
```

Comment:
We can set this up exactly like we did projects, get Max QuoteNumber, increase by 1.